packages/components/src/resizable-box/resize-tooltip/utils.ts

```diff
@@ -142,8 +140,8 @@ export function useResizeLabel( {
 		}
 
 		onResize( { width, height } );
-		debounceUnsetMoveXY();
-	}, [ width, height ] );
+		unsetMoveXY();
```

Comment:
The renaming to `unsetMoveXY` implicitly also makes it harder for a developer reading this code to notice the "debounce" logic  — we could either keep the `debounceUnsetMoveXY` function name and/or add an inline comment inside the `unsetMoveXY` to highlight the fact that a debounce technique is being used to hide the tooltip only after the user has stopped resizing the component.

Reply:
Another variation of the first approach above could be to also keep the original `unsetMoveXY` function, nbut just moving it inside `debounceUnsetMoveXY`:

```diff
diff --git a/packages/components/src/resizable-box/resize-tooltip/utils.ts b/packages/components/src/resizable-box/resize-tooltip/utils.ts
index 388bc41a84..7c122c2ebd 100644
--- a/packages/components/src/resizable-box/resize-tooltip/utils.ts
+++ b/packages/components/src/resizable-box/resize-tooltip/utils.ts
@@ -84,12 +84,8 @@ export function useResizeLabel( {
 	 */
 	const moveTimeoutRef = useRef< number >();
 
-	const unsetMoveXY = useCallback( () => {
-		if ( moveTimeoutRef.current ) {
-			window.clearTimeout( moveTimeoutRef.current );
-		}
-
-		moveTimeoutRef.current = window.setTimeout( () => {
+	const debounceUnsetMoveXY = useCallback( () => {
+		const unsetMoveXY = () => {
 			/*
 			 * If axis is controlled, we will avoid resetting the moveX and moveY values.
 			 * This will allow for the preferred axis values to persist in the label.
@@ -97,7 +93,13 @@ export function useResizeLabel( {
 			if ( isAxisControlled ) return;
 			setMoveX( false );
 			setMoveY( false );
-		}, fadeTimeout );
+		};
+
+		if ( moveTimeoutRef.current ) {
+			window.clearTimeout( moveTimeoutRef.current );
+		}
+
+		moveTimeoutRef.current = window.setTimeout( unsetMoveXY, fadeTimeout );
 	}, [ fadeTimeout, isAxisControlled ] );
 
 	useEffect( () => {
@@ -140,8 +142,8 @@ export function useResizeLabel( {
 		}
 
 		onResize( { width, height } );
-		unsetMoveXY();
-	}, [ width, height, onResize, unsetMoveXY ] );
+		debounceUnsetMoveXY();
+	}, [ width, height, onResize, debounceUnsetMoveXY ] );
 
 	const label = getSizeLabel( {
 		axis,
```

Reply:
Feel free to follow the approach that you prefer!

Reply:
I see your point, and I like your approach. It's definitely more clear. Thanks for the diff!